haystack/document_store/elasticsearch.py

```diff
@@ -967,6 +974,29 @@ def delete_documents(self, index: Optional[str] = None, filters: Optional[Dict[s
         if self.refresh_type == "wait_for":
             time.sleep(2)
 
+    def get_document_without_embedding_count(self, filters: Optional[Dict[str, List[str]]] = None,
```

Comment:
Let's not create a new method here but rather extend the existing `get_document_count()` by a new bool arg `only_documents_without_embedding` . Similar to `_get_all_documents_in_index()`. IMO that's the cleaner approach. 

Eventually, we will probably refactor our `filters` syntax to cope with this case directly, but until now the extra arg seems okay.

FYI: While having a look at the current implementation of `_get_all_documents_in_index()`, I think I spotted a bug (#1082 ). Let's not copy that bug to your new method here ;)

Reply:
@tholor Ok. I will add new arg in get_document_count() and update the test cases to support the latest change.

Yeah, Filter implementation needs to improve to support different scenarios.
 I will fix bug #1082 too in the same pull request.

Reply:
Awesome. Thanks